shells/backends/gcp_tf_backend/src/driver.py

```diff
@@ -0,0 +1,295 @@
+        # return response
+        return len(response) > 0
+
+    def _generate_state_file_string(self, gcp_backend_resource: GcpTfBackend):
+
+        tf_state_file_string = f'terraform {{\n\
+\tbackend "gcs" {{\n\
+\t\tbucket = "{gcp_backend_resource.bucket_name}"\n\
+\t\tprefix = "{gcp_backend_resource.prefix}"\n\
+\t}}\n\
+}}'
+        return tf_state_file_string
+
+
+    def _create_gcp_session(self, context, project):
+        with LoggingSessionContext(context) as logger:
+            api = CloudShellSessionContext(context).get_api()
+            gcp_backend_resource = GcpTfBackend.create_from_context(context)
+            try:
+                private_key = api.DecryptPassword(gcp_backend_resource.private_key).Value.replace("\\n", "\n")
+                email = api.DecryptPassword(gcp_backend_resource.client_email).Value
+                project_id = project
+                if not project_id:
+                    self._handle_exception_logging(logger, "Project id must be filled")
+                # Keys defines on AWS TF BACKEND RESOURCE
+                if private_key and email:
+                    if gcp_backend_resource.cloud_provider:
+                        self._handle_exception_logging(logger, "Only one method of authentication should be filled")
+                    # bucket_name = gcp_backend_resource.bucket_name
+                    with open(DYNAMIC_JSON, 'w', encoding='utf-8') as f:
+                        json.dump({ "type": TYPE_ACCOUNT, "project_id": project_id, "private_key": private_key, "client_email": email,"auth_uri": AUTH_URI, "token_uri": TOKEN_URI }, f, ensure_ascii=False, indent=4)
+                    os.environ[GOOGLE_APPLICATION_CREDENTIALS] = DYNAMIC_JSON
+                # Keys not defines on AWS TF BACKEND RESOURCE (CLP reference should have been set)
+                else:
+                    # CLP had not been set...
+                    if not gcp_backend_resource.cloud_provider:
+                        self._handle_exception_logging(logger, "At least one method of authentication should be filled")
+                    # Check a correct CLP has been reference
+                    # clp_details = api.GetResourceDetails(aws_backend_resource.cloud_provider)
+                    # print(f"gcp_backend_resource: {gcp_backend_resource.cloud_provider}")
+                    # self._handle_exception_logging(logger, f"gcp_backend_resource: {gcp_backend_resource.cloud_provider}")
+                    # logger.exception(f"gcp_backend_resource: {gcp_backend_resource.cloud_provider}")
+                    # if gcp_backend_resource.cloud_provider == GCP1G_MODEL:
+                    # GCP1G_MODEL:
+                    clp_details = self._validate_clp(api, gcp_backend_resource, logger)
+                    if clp_details.ResourceModelName == GCP1G_MODEL:
+                        print(clp_details.ResourceModelName)
+                        myactual=api.GetResourceDetails(clp_details.Name)
```

Comment:
you already have the resource details, why get the details again for the same resource?